### ISSUE TITLE
Call multiple beacon nodes in parallel

### ISSUE THREAD

**Issue by corverroos** [ObolNetwork/charon#960]
## Problem to be solved

eth2-go-client support multiple beacon nodes, but it does "failover" sequentially. This helps when calls to beacon node sporadically fail quickly, but it doesn't help if calls hang or are very slow.

A lot beacon node calls timeout sporadically if using infura or even local beacon nodes. Currently adding additional beacon nodes doesn't help to address this issue.

## Proposed solution

Instead of doing fancy "which beacon node is active" and sequential failover like current go-eth2-client.

Do more of a "spray-and-pray" approach. Fire all calls to all beacon nodes, and return the first non-error result. 

If all results error, return the first one. If the input context was cancelled, return context cancelled.

This involves adding our own multi-beacon-node support to eth2wrap. Take the opportunity to make that eth2wrap.Service our main eth2 instance (instead of using eth2client.Service which needs to be cast every time).

### Out of Scope

If there is anything to highlight as out of scope for this issue, please outline it here.
